Modules/CommandModule.cs

```diff
@@ -0,0 +1,305 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+
+using BrackeysBot.Commands;
+
+namespace BrackeysBot.Modules
+{
+    /// <summary>
+    /// Provides a module to execute commands.
+    /// </summary>
+    public class CommandModule
+    {
+        public CommandService Service => _commandService;
+        private CommandService _commandService;
+
+        public IServiceProvider ServiceProvider { get; set; }
+        public string CommandPrefix { get; set; }
+
+        private DiscordSocketClient _client;
```

Comment:
I don't feel like the CommandModule should have an instance of the client. What I suggest is to just subsribe to the HandleCommand class from outside `CommandModule`.

Also, the name `CommandModule` isn't really correct since that class isn't a command. Should be renamed to `CommandHandler` or something similar.